farm/data_handler/processor.py

```diff
@@ -454,13 +465,17 @@ def __init__(
                           metric=metric,
                           label_list=label_list,
                           label_column_name=label_column_name,
+                          text_column_name=text_column_name,
                           task_type=task_type)
         else:
             logger.info("Initialized processor without tasks. Supply `metric` and `label_list` to the constructor for "
                         "using the default task or add a custom task later via processor.add_task()")
 
     def file_to_dicts(self, file: str) -> [dict]:
-        column_mapping = {task["label_column_name"]: task["label_name"] for task in self.tasks.values()}
+        column_mapping = {}
+        for task in self.tasks.values():
+            column_mapping[task["label_column_name"]] = task["label_name"]
+            column_mapping[task["text_column_name"]] = "text"
```

Comment:
This might lead to unforeseen problems if a column "text" already exists.
I know we also do not check for the existence of task["label_name"] but now could be a good time to improve.

What about we check if any of task["label_name"] or "text" is already inside the dataframe and log a warning?

Reply:
Good point. I will come up with a solution...

Reply:
Ok - so `column_mapping` is a dict which maps "col_name_in_tsv_file" to "col_name_farm_wants". 
In `read_tsv` (`utils.py`) **all** of the tsv file is loaded with pandas. Then this is done:

```python
    columns = list(rename_columns.keys())
    df = df[columns]
```

This takes the keys from the dict as a list. The keys are the "col_name_in_tsv_file". So after that I have a pandas frame with only those cols I want to have. Then they are renamed: 
`df.rename(columns=rename_columns, inplace=True)`.

It the dict maps from `text` to `text` that will be no problem. It just selects the text col and then renames `text` to `text`. This renaming is nonsence but works. It could be removed but creates more code.

Before calling `df.rename(columns=rename_columns, inplace=True)` I could remove all entries from the dict where the key is the same as the value. Should I do that? What do you think? IMO it is not necessary.

But while discussing this it brought me to an other idea:

In `pd.read_csv` I could only read only those cols I need. Those in `list(rename_columns.keys())`. That would have the advantage that the code crashes early when the user mistyped a col name that is not in the data.

Reply:
Thanks for this detailed analysis. Your proposed changes make sense and improve upon functionality.